organization/search_indexes.py

```diff
@@ -0,0 +1,205 @@
+        'open_ended_s': open_ended,
+        'organization_name_s': name,
+        'organization_parent_name_ss': parent_names,
+        'organization_parent_count_i': parent_count,
+        'organization_membership_ss': memberships,
+        'organization_classification_ss': classes,
+        'organization_classification_count_i': class_count,
+        'organization_alias_ss': aliases,
+        'organization_headquarters_s': hq,
+        'organization_exact_location_ss': exactloc_names,
+        'organization_site_count_i': last_site_exists,
+        'organization_country_count_i': country_count,
+        'organization_area_ss': areas,
+        'organization_start_date_dt': first_cited,
+        'organization_end_date_dt': first_cited,
+        'organization_adminlevel1_ss': list(admin_l1_names),
+        'text': content
+    }
+    '''
+
+    CONTENT_FIELDS = (
+        'name',
+        'aliases',
+        'classifications',
+        'headquarters',
+        'parent_names',
+        'countries',
+        'exact_locations',
+        'areas',
+    )
+
+    open_ended = indexes.CharField()
+    name = indexes.CharField()
+    parent_names = indexes.MultiValueField(faceted=True)
+    memberships = indexes.MultiValueField(faceted=True)
+    classifications = indexes.MultiValueField(faceted=True)
+    aliases = indexes.MultiValueField()
+    headquarters = indexes.CharField()
+    exact_locations = indexes.MultiValueField()
+    areas = indexes.MultiValueField()
+    adminlevel1s = indexes.MultiValueField(faceted=True)
+
+    def get_model(self):
+        # For some reason, Haystack doesn't want to discover the index if we
+        # import the model into the global namespace for this module. So,
+        # import it here.
+        from organization.models import Organization
+
+        return Organization
+
+    def prepare(self, object):
+        self.prepared_data = super().prepare(object)
+
+        self.prepared_data['content'] = self._prepare_content(self.prepared_data)
+        self.prepared_data['countries'] = self._prepare_countries(self.prepared_data)
+
+        return self.prepared_data
+
+    def _prepare_countries(self, prepared_data):
+        countries = set()
+
+        for division in prepared_data['division_ids']:
+            countries.update([country_name(division)])
+
+        return list(countries)
+
+    def prepare_published(self, object):
+        parents = object.parent_organization.all()
+
+        return all([p.value.published for p in parents])
+
+    def prepare_division_ids(self, object):
+        division_ids = set()
+
+        # Start by getting the division ID recorded for this org
+        org_division_id = object.division_id.get_value()
+        if org_division_id:
+            division_ids.update([org_division_id.value])
+
+        # Grab foreign key sets
+        emplacements = object.emplacementorganization_set.all()
+
+        for emp in emplacements:
+            emp = emp.object_ref
```

Comment:
this is a super confusing variable override.

can we have 

`emp_object = emp.object_ref`